### PR TITLE
Safe JSON serializer & content-length fix (issue with Routes in Revit French)

### PR DESCRIPTION
**Pull request by vinnividivicci**
Safe JSON serializer & content-length fix (issue with Routes in Revit French)

## Description

Add a Unicode-safe JSON serializer (_safe_json_dumps) to handler.py to avoid json.dumps UnicodeDecodeError on IronPython by escaping non-ASCII as \uXXXX. Use json.dumps(..., ensure_ascii=True) with a fallback to _safe_json_dumps when encoding errors occur. Update HttpRequestHandler._write_response to always compute a bytes body, set a correct Content-Length header, skip duplicate Content-Length from response headers, and write the encoded body. Add unit tests covering response framing (Content-Length and body bytes) and Unicode-safe JSON serialization/parse_response behavior.

---

## Checklist

Before submitting your pull request, ensure the following requirements are met:

- [x] Code follows the [PEP 8](https://peps.python.org/pep-0008/) style guide.
- [x] Code has been formatted with [Black](https://github.com/psf/black) using the command:
  ```bash
  pipenv run black {source_file_or_directory}
  ```
- [x] Changes are tested and verified to work as expected.

---

## Related Issues

If applicable, link the issues resolved by this pull request:

- Resolves #[issue number]

---

## Additional Notes

Include any additional context, screenshots, or considerations for reviewers.

---

Thank you for contributing to pyRevit! 🎉

### CODE REVIEW

**Comment by vinnividivicci** [comment]
@jmcouffin Bug fixes to the routes API to work properly with Revit, French, and other multilingual Revit installations with non English characters.

File: pyrevitlib/pyrevit/routes/server/server.py [hunk 1]
```diff
@@ -179,58 +185,56 @@ def _process_request(self, method):
...
     # delete
     def do_DELETE(self):
-        self._process_request(method='DELETE')
+        self._process_request(method="DELETE")
 
     # rest of standard http methods -------------------------------------------
     # https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
     def do_HEAD(self):
-        self._process_request(method='HEAD')
+        self._process_request(method="HEAD")
```
[the remark this answers is not in the document]
Comment (vinnividivicci) [reply pyrevitlib/pyrevit/routes/server/server.py:220]: The co-pilot review erroneously picked up on this, but all we did in this pull request for this piece of the code is change the single quotes to double quotes. So I will ignore this comment. Maybe it can be dealt with later, but is out of the scope of this pull request.

File: pyrevitlib/pyrevit/routes/server/server.py [hunk 2]
```diff
@@ -121,12 +121,21 @@ def _call_host_event_sync(self, req_hndlr, event_hndlr):
...
         if r.headers:
             for key, value in r.headers.items():
+                if str(key).lower() == "content-length":
+                    continue
                 self.send_header(key, value)
-            self.end_headers()
+        self.end_headers()
         # sending \n if no data otherwise Postman panics for some reason
-        self.wfile.write(r.data or "\n")
+        self.wfile.write(body)
```
[the remark this answers is not in the document]
Comment (vinnividivicci) [reply pyrevitlib/pyrevit/routes/server/server.py:138]: We kept the existing approach to make it backwards compatible. So this is out of scope.

**Comment by vinnividivicci** [comment]
Added test menu for some of the routes code that I touched (additional coverage needed in future for pre-existing code):
image

Tests pass:
image

**Comment by jmcouffin** [comment]
Thanks @vinnividivicci
